File: sample-conference-videochat/sample-conference-videochat/Components/Settings/SettingSwitchCell.m
//
//  SettingSwitchCell.m
//  sample-videochat-webrtc
//
//  Created by Andrey Ivanov on 30/09/15.
//  Copyright © 2015 QuickBlox Team. All rights reserved.
//

#import "SettingSwitchCell.h"
#import "SwitchItemModel.h"

@interface SettingSwitchCell()

@property (weak, nonatomic) IBOutlet UISwitch *switchCtrl;

@end

@implementation SettingSwitchCell

- (void)setModel:(SwitchItemModel *)model {
    
    [super setModel:model];
    self.switchCtrl.on = model.on;
}

- (IBAction)valueChanged:(UISwitch *)sender {
    
    SwitchItemModel *model = (id)self.model;
    model.on = sender.on;
    [self.delegate cell:self didChageModel:self.model];
}

@end

File: sample-conference-videochat-swift/sample-conference-videochat-swift/sample-conference-videochat-swift/Components/Settings/SettingSwitchCell.swift
//
//  SettingSwitchCell.swift
//  sample-conference-videochat-swift
//
//  Created by Vladimir Nybozhinsky on 22.10.2018.
//  Copyright © 2018 QuickBlox. All rights reserved.
//

import UIKit

class SettingSwitchCell: BaseSettingsCell {

    @IBOutlet private weak var switchCtrl: UISwitch!
    var modelSwitch = SwitchItemModel()
    
    override func updateModel(_ model: BaseItemModel) {
        modelSwitch = model as! SwitchItemModel
        switchCtrl.isOn = modelSwitch.on
        label.text = modelSwitch.title
    }
    
    @IBAction func valueChanged(_ sender: UISwitch) {
        modelSwitch.on = sender.isOn
        delegate?.cell(self, didChageModel: modelSwitch)
    }
}
